components/batch-system/src/batch.rs

```diff
@@ -463,6 +484,15 @@ impl<N: Fsm, C: Fsm, Handler: PollHandler<N, C>> Poller<N, C, Handler> {
                 batch.schedule(&self.router, *offset, true);
             }
             to_skip_end.clear();
+            if !run && batch.control.is_some() {
+                let len = self.handler.handle_control(batch.control.as_mut().unwrap());
+                if batch.control.as_ref().unwrap().is_stopped() {
+                    batch.remove_control(&self.router.control_box);
+                } else if let Some(len) = len {
+                    batch.release_control(&self.router.control_box, len);
```

Comment:
Why not release control fsm outside the loop?

Reply:
Fixed, thanks!